### DISCUSSION TITLE
App UI reactivity - single table row

### DISCUSSION THREAD

**Discussion by gcleaves** [windmill-labs/windmill#2779] [Questions] [answered]
Hi, can you please help me understand the reactivity model of UI components. My goal is to update a single table row after a successful database update. I've tried `table.selectedRow = {a:1,b:2}` but this doesn't seem to work. 

I know there is a `setValue(id:string, value)` function but I'm not sure of the correct way to update a single table row. Is it possible?

Thanks as always for your help.

Geoff

**Comment by rubenfiszel** [comment, marked as answer]
The Reactivity model that we use is based on the concepts of Reactive Programming. https://en.wikipedia.org/wiki/Reactive_programming

UI reactivity is based on dataflows and it's often not possible to mutate values directly except for mutable components by nature such as inputs. We do have one mutable data source which is state. So you can either re-read completely your table after update OR make your table use as datasource state.foo and mutate state.foo directly with any transformer/frontend scripts. You will need a script to also hydrate initially state.foo

The reason for avoiding mutability is that it's an anti-pattern for highly reactive, large scale apps. Let say you mutate your table and then one of the input of the runnable your table is based change and trigger a refresh of it, now your mutation is gone.
